### DISCUSSION TITLE
Subdomain tenancy

### DISCUSSION THREAD

**Discussion by josefbehr** [filamentphp/filament#5921] [Ideas]
I'd love to use v3 for a multi-tenant saas app using subdomains for tenant identification. Currently that's not easily possible, as e.g. getLoginUrl() in the HasAuth trait generates the URL without the then-required tenant parameter.
I'd implement a solution, but not sure how you would like to approach this. So let me know and I'll get to work :)

**Comment by danharrin** [comment]
Would it not be a case of adding an option to replace this line `path()` with `domain()`?

![Screenshot 2023-03-13 at 18 24 24](https://user-images.githubusercontent.com/41773797/224795114-96431b6e-572e-48b6-a2a7-319d624d1a73.png)

**Reply by danharrin** [reply]
And because (I presume) you can have a parameter within the domain, the rest of the URL generation should work as the only thing that changes is where the parameter is within the route

**Reply by josefbehr** [reply]
![image](https://user-images.githubusercontent.com/4368880/224838397-a7268a47-c085-4525-8507-0c66c075370a.png)

getLoginUrl and other methods in HasRoutes call route('filament.{$contextId}.auth.login') and so on without parameters, so the URLs are missing that parameter for the tenant context with subdomains. Not sure if I am allowed to adapt the methods there making them tenancy aware as that would create a dependency on a separate trait (the HasTenancy one).

**Reply by danharrin** [reply]
Why `{tenantslug}`? Shouldn't it just be `{tenant}`?

**Reply by josefbehr** [reply]
To do that, we would also need to receive the current tenant somehow. At the moment, this is not stored as far as I can see.

**Reply by danharrin** [reply]
The current tenant is set in middleware and attached to the FilamentManager singleton. The process should be identical with subdomain or subdirectory routing I think

**Reply by josefbehr** [reply]
> Why `{tenantslug}`? Shouldn't it just be `{tenant}`?

Yes, but it doesn't matter, problem persists.

**Reply by josefbehr** [reply]
> The current tenant is set in middleware and attached to the FilamentManager singleton. The process should be identical with subdomain or subdirectory routing I think

Ok, let me try to get something together quickly.

**Reply by josefbehr** [reply]
The tenant is only set after authentication in the Authenticate middleware. Maybe I can move that to an earlier place so registration and login can be tenant-specific.

**Reply by danharrin** [reply]
some checks will need to remain in Authenticate to ensure the auth'd user has access to the tenant

**Reply by josefbehr** [reply]
I'll create a draft PR for you to check. I think there's other improvements regarding tenancy that can be done, but for now this should work for basic subdomain tenancy.

**Reply by josefbehr** [reply]
https://github.com/filamentphp/filament/pull/5927

any feedback is appreciated
